h/services/jwt.py

```diff
@@ -47,7 +106,89 @@ def decode_token(
             )
         except PyJWTError as err:
             msg = f"Invalid JWT for: {iss}, {aud}. {err}"
-            raise TokenValidationError(msg) from err
+            raise JWTDecodeError(msg) from err
+
+    def encode(
+        self,
+        payload: dict,
+        issuer: str,
+        audience: str,
+        *,
+        _exp=True,
+        _algorithm="HS256",
+        _signing_key=None,
+    ) -> str:
+        """Return the given `payload` in a signed JWT."""
+        signing_key = _signing_key or self.jwt_signing_key
+
+        if _exp:
+            payload["exp"] = datetime.now(tz=UTC) + timedelta(hours=1)
+
+        if issuer:
+            payload["iss"] = issuer
+
+        if audience:
+            payload["aud"] = audience
+
+        return jwt.encode(
+            payload,
+            key=None if _algorithm == "none" else signing_key,  # type: ignore[arg-type]
+            algorithm=_algorithm,
+        )
+
+    def decode(self, token: str, issuer: str, audience: str) -> dict:
+        """Decode the given `token` and return the original payload.
+
+        Decodes tokens from the JWTService.encode() method above.
+
+        :raise JWTDecodeError: if decoding the JWT fails
+        """
+        try:
+            payload = jwt.decode(
+                token,
+                self.jwt_signing_key,
+                algorithms=["HS256"],
+                options={"require": ["exp", "iss", "aud"]},
+                issuer=issuer,
+                audience=audience,
+            )
+        except InvalidTokenError as err:
+            raise JWTDecodeError from err
+
+        del payload["exp"], payload["iss"], payload["aud"]
+        return payload
```

Comment:
This is a pair of generic `encode()` and `decode()` methods that are well-tested and that attempt to implement some general best practices, e.g. using expiry, issuer and sub correctly and handling errors correctly. 

While these won't apply to all JWTs, are probably general enough to be used for many cases. Nonetheless, these should probably have a more specific name than just `encode()` and `decode()`. For example these do symmetric signing and don't support asymmetric keys/algorithms.